File: Java/longest-substring-without-repeating-characters.java
import java.util.HashMap;
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character, Integer> cIm= new HashMap<>();
        int l=0;
        int max=0;
        for(int r=0;r<s.length();r++){
            char act=s.charAt(r);
            if(cIm.containsKey(act)&&cIm.get(act)>=l){
                l=cIm.get(act)+1;
            }
            cIm.put(act,r);
            max=Math.max(max, r-l+1);
        }
        return max;
    }
}

File: Python/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        cIm={}
        l =0 
        maxl=0 #max length
      #iterates the string length
        for r in range(len(s)):
            if s[r]in cIm and cIm[s[r]]>=l:
                l=cIm[s[r]]+1
            cIm[s[r]]=r
            maxl=max(maxl,r-l+1)
        return maxl
